obelisk/cpp/obelisk_cpp/include/obelisk_sensor.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "obelisk_node.h"
+
+namespace obelisk {
+    class ObeliskSensor : public ObeliskNode {
+      public:
+        explicit ObeliskSensor(const std::string& name) : ObeliskNode(name) { has_sensor_pub_ = false; }
+
+        rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn virtual on_configure(
+            const rclcpp_lifecycle::State& prev_state) {
+            ObeliskNode::on_configure(prev_state);
+
+            // TODO (@zolkin): Use the MESSAGE_NAME field (stored in info.msg_type in registered_publishers_) to
+            // determine if the publisher is publishing a sensor
+            //  for now, just going to check if its empty
```

Comment:
is there some way to do what the python side does here? we're just checking whether at least one of the publisher message types is a sensor message, and if not, we throw a runtime error

[https://github.com/Caltech-AMBER/obelisk/blob/6a4dc6259b04a9d192f6eb90435e0aab88ac85ed/obelisk/python/obelisk_py/core/sensing.py](https://github.com/Caltech-AMBER/obelisk/blob/6a4dc6259b04a9d192f6eb90435e0aab88ac85ed/obelisk/python/obelisk_py/core/sensing.py)

Reply:
So if we want to manage a list of all the allowed types somewhere, then I can do this. I think we want to move away from this in the long run, but for now I can add it in.